keeper-gradle-plugin/src/main/java/com/slack/keeper/InferAndroidTestKeepRules.kt

```diff
@@ -126,18 +154,24 @@ public abstract class InferAndroidTestKeepRules : JavaExec() {
       }
 
       group = KEEPER_TASK_GROUP
-      androidTestJar.set(androidTestJarProvider.flatMap { it.archiveFile })
-      appJar.set(releaseClassesJarProvider.flatMap { it.archiveFile })
-      classpathJar.set(androidJar)
+      androidTestSourceJar.set(androidTestJarProvider.flatMap { it.archiveFile })
+      appTargetJar.set(releaseClassesJarProvider.flatMap { it.archiveFile })
+      androidLib.set(androidJar)
+      androidTestLib.set(androidTestJar)
       jvmArgsProperty.set(extensionJvmArgs)
+      this.traceReferencesEnabled.set(traceReferencesEnabled)
+      this.traceReferencesArgs.set(traceReferencesArgs)
       outputProguardRules.set(
           project.layout.buildDirectory.file(
               "${KeeperPlugin.INTERMEDIATES_DIR}/inferred${variantName.capitalize(
                   Locale.US)}KeepRules.pro"))
       classpath(r8Configuration)
-      main = "com.android.tools.r8.PrintUses"
+      main = when (this.traceReferencesEnabled.get()) {
```

Comment:
I believe there's a mainClass property we could use now and set it to a mapped property. Something like this

```kotlin
mainClass.set(traceReferencesEnabled.map { enabled -> 
  if (enabled) "..." else "..."
})
```

Reply:
```
* What went wrong:
Execution failed for task ':inferInternalReleaseAndroidTestKeepRulesForKeeper'.
> The value for task ':inferInternalReleaseAndroidTestKeepRulesForKeeper' property 'mainClass' is final and cannot be changed any further.
```
Seems the timing is as important for main and mainClass.

Reply:
If we use the property then timing shouldn't be an issue. Can you push a change with it using just `mainClass` so I can see what you're trying to do?

Reply:
I used the `mainClass` and moved the instruction in the task's exec function.
If you want to investigate the reason why, maybe the easiest would be to try yourself locally by update/move the line 🙇